mqtt/mqtt-edgehub/tests/authorization.rs

```diff
@@ -19,6 +20,169 @@ use mqtt_edgehub::{
 mod common;
 use common::DummyAuthorizer;
 
+/// Scenario:
+/// create broker
+/// create command handler
+/// connect authorized client
+/// verify client has not been connected, since authorized identities haven't been sent
+#[tokio::test]
+async fn publish_not_allowed_identity_not_in_cache() {
+    // Start broker with DummyAuthorizer that allows everything from CommandHandler and $edgeHub,
+    // but otherwise passes authorization along to EdgeHubAuthorizer
+    let broker = BrokerBuilder::default()
+        .with_authorizer(DummyAuthorizer::new(EdgeHubAuthorizer::new(
+            authorize_fn_ok(|activity| {
+                if matches!(activity.operation(), Operation::Connect(_)) {
+                    Authorization::Allowed
+                } else {
+                    Authorization::Forbidden("not allowed".to_string())
+                }
+            }),
+        )))
+        .build();
+    let broker_handle = broker.handle();
+
+    let server_handle = start_server(broker, DummyAuthenticator::with_id("device-1"));
+
+    // start command handler with AuthorizedIdentitiesCommand
+    let command = AuthorizedIdentitiesCommand::new(&broker_handle);
+    let (command_handler_shutdown_handle, join_handle) =
+        common::start_command_handler(server_handle.address(), command)
+            .await
+            .expect("could not start command handler");
+
+    let mut device_client = PacketStream::connect(
+        ClientId::IdWithCleanSession("device-1".into()),
+        server_handle.address(),
+        None,
+        None,
+    )
+    .await;
+
```

Comment:
nit:
```
assert_matches!(device_client.next().await, Some(Packet::ConnAck(c)) if c.return_code == ConnectReturnCode::Accepted);
```

Reply:
ah nice this is what I was trying to go for